Removing development team from Xcode project.

Rojuinex opened the pull request:
Fixes #16881 

Removing development team from Xcode project.

Test Plan:
----------
* Changed application's react-native dependency to point to my github repo
* Deleted `node_modules/react-native`
* `npm install`
* Build and Archived the project with Xcode and Xcode Server

Release Notes:
--------------
[IOS] [ENHANCEMENT] [React/React.xcodeproj/project.pbxproj] - Removed Development Team ID from Xcode project

rawit1976 commented:
Good

Rojuinex commented:
Not exactly sure why the e2e test is failing but it seems to have something to do with packager?  Log doesn't seem to provide more info than "Test operation was canceled".  Anyone have any thoughts?

Rojuinex commented:
@GuoZhiQiang Have you added your own development team to your project?

GuoZhiQiang commented:
@Rojuinex Not yet, I just run my project  in simulator. So i don't add my development team to the project

Rojuinex commented:
@GuoZhiQiang I think I may have fixed that issue.  I updated the pull request and removed automatic signing from the project settings which I think is why it was crashing.  Can you try updating?
